Active record basics

Krutik31 opened the pull request:
Applied CRUD operations on the Product model in console. 
Added simple validations & callbacks.
Used bootstrap for designing index & show pages.

Krutik31 committed:
feat: TE7-T116 Active Record Basics

Applied CRUD operations on the Product model in console.
Added simple validations & callbacks.
Used bootstrap for designing index & show pages.

Krutik31 committed:
Update ActiveRecord-README.md

purvak-simform commented on config/routes.rb:
```diff
@@ -1,7 +1,9 @@
-  get 'home/index'
+  get '/product/', to: "product#index"
+  get '/home/', to: "home#index"
+  get 'product/:id', to: "product#show"
```
@Krutik31 There is no need to create manual routes for product. You can write it like this
  resources :product, only: [:index, :show]

Krutik31 replied:
Changes made on routes

purvak-simform replied:
@Krutik31  in routes controller name are always plural

purvak-simform commented on app/views/product/index.html.erb:
```diff
@@ -0,0 +1,26 @@
+          <td><%= product.id %></td>
+          <td><%= product.title %></td>
+          <td>
+            <a href="/product/<%= product.id %>" class="btn btn-sm btn-outline-success">View</a>
```
@Krutik31 You are creating a link here, so use rails method link_to
Check it here how can a link be created in rails
https://www.rubyguides.com/2019/05/rails-link_to-method/

purvak-simform commented on app/views/product/show.html.erb:
```diff
@@ -0,0 +1,33 @@
+    <%= @product.brand %>
+  </h5><hr>
+  
+  <a class="btn btn-flat btn-danger float-left" href="/product">Back</a>
```
Same as above

Krutik31 committed:
perf: TE7-T116 Active Record Basics

Done some changes to the routes

purvak-simform commented on app/models/product.rb:
```diff
@@ -0,0 +1,22 @@
+  after_commit :commitedmsg
+  def commitedmsg
+    puts "Message after successful commiting from 'after_commit' callback."
+  end
```
@Krutik31 We don't write callback then method, write all callbacks together and methods downside
And take a caution while keeping methods name it should be meaningful
`commitedmsg` instead change it to something like `display_commit_callback_message`

Similarly change for above

Krutik31 committed:
perf: TE7-T116 Active Record Basics

Made suggested changes.
updated html files & followed naming conventions in callbacks.

Krutik31 committed:
refactor: TE7-T116 Active Record Basics

Followed Naming conventions in controller and made changes in other files accordingly.
